bugsnag-plugin-android-ndk/src/main/jni/utils/serializer.c

```diff
@@ -148,11 +147,31 @@ bugsnag_event *bsg_map_v2_to_report(bugsnag_report_v2 *report_v2) {
     // Fatal C errors are always true by default, previously this was hardcoded and
     // not a field on the struct
     event->unhandled = true;
+    free(report_v2);
   }
-  free(report_v2);
   return event;
 }
 
+void migrate_breadcrumb_legacy(bugsnag_report_v2 *report_v2, bugsnag_event *event) {
+  for (int k = 0; k < report_v2->crumb_count; k++) {
+    int crumb_index = report_v2->crumb_first_index + k % BUGSNAG_CRUMBS_MAX;
```

Comment:
As it's a circular structure, do we need to start again at zero until you've gone through all the crumbs?

Reply:
Added parentheses to this expression, otherwise I believe this should start the index at 0 once the value is `>= BUGSNAG_CRUMBS_MAX`